katello/katello-change-hostname

```diff
@@ -50,108 +124,105 @@ def yesno
-  exit(false)
+  fail_with_message("Hostname change aborted, no changes have been made to your system")
 end
 
-unless options[:username] && options[:password]
-  STDOUT.puts "Username and/or Password options are missing!"
-  puts opt_parser
-  exit(false)
+unless @options[:username] && @options[:password]
+  fail_with_message("Username and/or Password options are missing!", opt_parser)
 end
 
 if ARGV[0] && ARGV.count >= 1
   new_hostname = ARGV[0]
 else
-  puts opt_parser
-  exit(false)
+  fail_with_message("Please specify a hostname.", opt_parser)
 end
 
 STDOUT.puts "\nChecking overall health of server"
 unless hammer_ping_success
-  STDOUT.puts "There is a problem with the server, please check 'hammer ping'"
-  exit(false)
+  fail_with_message("There is a problem with the server, please check 'hammer ping'")
+end
+
+if @foreman_proxy_content
+  check_for_certs_tar
+  fpc_installer_args = get_fpc_answers
 end
 
 # Get the hostname from your system
 old_hostname = get_hostname
 
-scenario_answers = YAML.load_file("/etc/foreman-installer/scenarios.d/#{options[:scenario]}-answers.yaml")
+scenario_answers = YAML.load_file("/etc/foreman-installer/scenarios.d/#{@options[:scenario]}-answers.yaml")
 ssl_build_dir = scenario_answers["certs"]["ssl_build_dir"]
 ssl_backup_dir = "#{ssl_build_dir}-#{timestamp}-#{old_hostname}"
 
-STDOUT.puts "Updating default #{proxy}"
-default_capsule_id = `hammer -u #{options[:username]} -p #{options[:password]} capsule list | grep #{old_hostname} | awk '{ print $1 }'`[0]
-# Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
-`hammer -u #{options[:username]} -p #{options[:password]} capsule update --id #{default_capsule_id} --url https://#{new_hostname}:9090 --new-name #{new_hostname} 2> /dev/null`
-
-STDOUT.puts "Updating installation media paths"
-installation_media_ids = `hammer -u #{options[:username]} -p #{options[:password]} medium list | grep "#{old_hostname}" | awk '{ print $1 }'`.split("\n")
-installation_media_paths = `hammer -u #{options[:username]} -p #{options[:password]} medium list | grep "#{old_hostname}" | awk '{ print $5 }'`.split("\n")
-installation_media_ids.zip(installation_media_paths).each do |id, path|
-  `hammer -u #{options[:username]} -p #{options[:password]} medium update --id #{id} --path #{path.sub! old_hostname, new_hostname}`   
+unless @foreman_proxy_content
+  STDOUT.puts "Updating default #{@proxy}"
+  default_capsule_id = `hammer -u #{@options[:username]} -p #{@options[:password]} capsule list | grep #{old_hostname} | awk '{ print $1 }'`[0]
+  # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
+  `hammer -u #{@options[:username]} -p #{@options[:password]} capsule update --id #{default_capsule_id} --url https://#{new_hostname}:9090 --new-name #{new_hostname} 2> /dev/null`
+
+  STDOUT.puts "Updating installation media paths"
```

Comment:
I didn't think we created install media anymore. So why do we need to update them? I realize this is from the previous incarnation of the script just curious.